### ISSUE TITLE
Godot 4.0.3 RC2 exits immediately after clicking on a Skeleton3D node

### ISSUE THREAD

**Issue by uguu-org** [godotengine/godot#77046]
### Godot version

v4.0.3.rc2.official.2ac4e3bb3

### System information

Windows 10, NVIDIA GeForce GTX 745

### Issue description

Godot 4.0.3 RC2 exits immediately after clicking on a Skeleton3D node.  This does not happen with 4.0.3 RC1, so it appears to be a regression with RC2.

### Steps to reproduce

1. Open model.tscn in attached project.
2. Click on Skeleton3D node
3. Observe that Godot draw a box briefly, then exits before showing skeleton hierarchy.
![click_on_skeleton](https://github.com/godotengine/godot/assets/2000051/21a94e7f-3b5e-4722-89bc-7303613cf065)

Expected behavior (from RC1).  I am having trouble recording mouse cursor, but basically I clicked on Skeleton3D on the left and the bone hierarchy shows up on the right.

https://github.com/godotengine/godot/assets/2000051/a97ad85e-7059-4a5c-92fb-03f3e748fdbe

Actual behavior (from RC2).  The yellow box shows up, and Godot exits abruptly, which causes the recorder to stop.

https://github.com/godotengine/godot/assets/2000051/60ab12c8-2903-4e5b-9b80-bf660713b2cc

### Minimal reproduction project

[skeleton_test.zip](https://github.com/godotengine/godot/files/11470841/skeleton_test.zip)

**Comment by lyuma** [comment]
cc @lyuma @fire @TokageItLab

**Comment by fire** [comment]
Looking into it too.

**Comment by fire** [comment]
Test plan:

1. recreate problem
2. get backtrace
3. git bisect the list of changes from rc1 to rc2

**Comment by akien-mga** [comment]
Probably https://github.com/godotengine/godot/pull/76592, CC @spanzeri

**Comment by uguu-org** [comment]
Attached is a smaller test project, click on Skeleton3D node to observe the abrupt exit.
[skeleton_test_2.zip](https://github.com/godotengine/godot/files/11472437/skeleton_test_2.zip)

![click_on_skeleton](https://github.com/godotengine/godot/assets/2000051/bedadcf6-dc04-47b3-bd50-90ab84f1c89a)

**Comment by spanzeri** [comment]
I'll have a look

**Comment by spanzeri** [comment]
https://github.com/godotengine/godot/pull/77074 this fixes the crash.
However, I am not sure about the fix at all.
As I mentioned in the PR, Control nodes seems to fire THEME_CHANGED notifications before propagating ENTER_TREE to derived classes.
Accessing theme resources (to create the editor) before enter tree has caused a lot of warnings everywhere else.
I am not completely familiar with how those system works yet (I am fairly new to the engine) and I could use someone with more expertise double checking what the correct sequence of events is (either the editor needs to be created at a different moment in time or something is not quite right with the way Controls handles ENTER_TREE)

**Comment by lyuma** [comment]
Some notes from rocket chat:
* The cause is almost certainly #76592

The part that was confusing to me was that is_inside_tree() returns true before _notification NOTIFICATION_ENTER_TREE has fired, but this is not new behavior. (the parent class is synchronously triggering another notification)

Before the PR #76592, widgets were being constructed in the constructor. from reading other editor code, this pattern is universal or very common

So by calling create_editors after the constructor, it breaks the pattern. Supposedly, some editor constructions were moved to ENTER_TREE, not just for this node. But I haven't seen this, and Yuri confirmed that it is rare to construct widgets in this way.

The other problem with #76592: if ENTER_TREE happens multiple times (reloading or reparenting) this will leak mem.

In summary, most editor components construct everything in the constructor, so they do not need to worry about ordering between ENTER_TREE and THEME_CHANGED. Otherwise, it might be necessary to use null check instead of is_inside_tree()
